File: pythondp - banking/model/account.py
import uuid
from enum import Enum


class AccountType(Enum):
    SAVINGS = "SAVINGS"
    CURRENT = "CURRENT"
    BUSINESS = "BUSINESS"


class AccountStatus(Enum):
    ACTIVE = "ACTIVE"
    FROZEN = "FROZEN"
    CLOSED = "CLOSED"


class Account:

    def __init__(self, user_id: str, account_type: AccountType):
        self.account_id = str(uuid.uuid4())
        self.user_id = user_id
        self.type = account_type
        self.status = AccountStatus.ACTIVE
        self.balance = 0.0

    def deposit(self, amount: float):
        self.balance += amount

    def withdraw(self, amount: float):
        if self.balance >= amount:
            self.balance -= amount
        else:
            raise Exception("Insufficient funds")

File: javadp - banking/model/Account.java
package model;

import java.util.UUID;

public class Account {

    private String accountId;
    private String userId;
    private double balance;
    private AccountStatus status;
    private AccountType type;

    public Account(String userId, AccountType type) {
        this.accountId = UUID.randomUUID().toString();
        this.userId = userId;
        this.type = type;
        this.status = AccountStatus.ACTIVE;
        this.balance = 0.0;
    }

    public void deposit(double amount) {
        this.balance += amount;
    }

    public void withdraw(double amount) {
        if (balance >= amount) {
            this.balance -= amount;
        } else {
            throw new RuntimeException("Insufficient funds");
        }
    }

    public double getBalance() {
        return balance;
    }

    public String getAccountId() {
        return accountId;
    }
}